### PR TITLE
STN Rules - after rebase

### PR DESCRIPTION
**Pull request by Batky**

### CODE REVIEW

**Comment by ondrej-fabry** [comment]
Lint fails with:
```
clientv1/defaultplugins/dbadapter/data_change_db.go:151:1:warning: comment on exported method PutDSL.StnRules should be of the form "StnRules ..." (golint)

plugins/defaultplugins/ifplugin/data_resync.go:237:1:warning: exported method StnConfigurator.Resync should have comment or be unexported (golint)

plugins/defaultplugins/ifplugin/stn_config.go:33:6:warning: exported type StnConfigurator should have comment or be unexported (golint)

plugins/defaultplugins/ifplugin/stn_config.go:58:1:warning: comment on exported function CheckStn should be of the form "CheckStn ..." (golint)

plugins/defaultplugins/ifplugin/stn_config.go:88:1:warning: exported method StnConfigurator.Add should have comment or be unexported (golint)

plugins/defaultplugins/ifplugin/stn_config.go:110:1:warning: exported method StnConfigurator.Delete should have comment or be unexported (golint)

plugins/defaultplugins/ifplugin/stn_config.go:137:1:warning: exported method StnConfigurator.Modify should have comment or be unexported (golint)

plugins/defaultplugins/ifplugin/vppcalls/stn_vppcalls.go:29:6:warning: exported type StnRule should have comment or be unexported (golint)

plugins/defaultplugins/ifplugin/vppcalls/stn_vppcalls.go:30:2:warning: struct field IpAddress should be IPAddress (golint)
```

File: clientv1/defaultplugins/dbadapter/data_change_db.go [hunk 1]
```diff
@@ -147,6 +148,12 @@ func (dsl *PutDSL) Arp(arp *l3.ArpTable_ArpTableEntry) defaultplugins.PutDSL {
 	return dsl
 }
 
+// StnRules adds a request to create or update VPP Access Control List.
```
Comment (VladoLavor) [inline clientv1/defaultplugins/dbadapter/data_change_db.go]: create or update VPP STN rule

File: clientv1/defaultplugins/dbadapter/data_change_db.go [hunk 2]
```diff
@@ -216,17 +222,21 @@ func (dsl *DeleteDSL) ACL(aclName string) defaultplugins.DeleteDSL {
 	return dsl
 }
 
+// StnRules adds Stn sules to the RESYNC request.
```
Comment (VladoLavor) [inline clientv1/defaultplugins/dbadapter/data_change_db.go]: rules

File: plugins/defaultplugins/ifplugin/bin_api/stn/stn.go [hunk 3]
```diff
@@ -0,0 +1,123 @@
+// Code generated by govpp binapi-generator DO NOT EDIT.
+// Package stn represents the VPP binary API of the 'stn' VPP module.
+// Generated from 'stn.api.json'
+package stn
```
Comment (VladoLavor) [inline plugins/defaultplugins/ifplugin/bin_api/stn/stn.go]: please verify that the file is generated against newest version of the VPP (vpp with commit id 4c9f2a805038a2d4f663b05a3d08ac4ee1eec3da), re-generate it if needed

File: plugins/defaultplugins/ifplugin/model/stn/keys_agent_stn.go [hunk 4]
```diff
@@ -0,0 +1,12 @@
+package stn
+
+// KeyPrefix returns the prefix used in ETCD to store vpp STN config
+func KeyPrefix() string {
+	return "vpp/config/v1/stn/rules/"
```
Comment (VladoLavor) [inline plugins/defaultplugins/ifplugin/model/stn/keys_agent_stn.go]: optionally you can define string constant "vpp/config/v1/stn/rules/" and return it here
then in func Key you can return that constant + ruleName

File: plugins/defaultplugins/ifplugin/stn_config.go [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+	} else {
+		plugin.Log.Warnf("Unregister failed, STN rule %v not found", stnID)
+	}
+
+	return nil
+}
+
+// Modify changes the stored rules
+func (plugin *StnConfigurator) Modify(rule *model_stn.StnRule, rule2 *model_stn.StnRule) error {
+	//TODO: Need to be implemented
```
Comment (VladoLavor) [inline plugins/defaultplugins/ifplugin/stn_config.go]: what about this? Are you going to implement it?

Comment (Batky) [reply plugins/defaultplugins/ifplugin/stn_config.go]: Goal is to use it on next patch for STN rules where rules are modified after interface changes.

File: clientv1/defaultplugins/dbadapter/data_change_db.go (hunk 1 above)
Comment (Batky) [reply clientv1/defaultplugins/dbadapter/data_change_db.go]: Rebase mistake.

**Pull request merged**